windows/src/buildtools/build_standards_data/Keyman.System.BuildLanguageSubtagRegistry.pas

```diff
@@ -22,6 +23,124 @@ implementation
 
 { TBuildLanguageSubtagRegistry }
 
+procedure TBuildLanguageSubtagRegistry.BuildSubtagRegistry(
+  const DestinationFile: string);
+var
+  FResult: TStringList;
+  FLanguages, FScripts, FRegions: TStringList;
+
+  procedure ParseList;
+  var
+    i, n: Integer;
+    line: string;
+    code: string;
+    value: string;
+    FType: string;
+    FID: string;
+    FDescription: string;
+  begin
+    for i := 0 to FSubtagRegistry.Count-1 do
+    begin
+      line := Trim(FSubtagRegistry[i]);
+      if line = '%%' then
+      begin
+        if Pos('..', FID) = 0 then
+        begin
+          if FType = 'language' then
+            FLanguages.Add(FID + '=' + FDescription)
+          else if FType = 'script' then
+            FScripts.Add(FID + '=' + FDescription)
+          else if FType = 'region' then
+            FRegions.Add(FID + '=' + FDescription);
+        end;
+        FType := '';
+        Continue;
+      end;
+
+      n := Pos(':', line);
+      if n = 0 then
+        Continue;
+
+      code := Copy(line, 1, n-1);
+      value := Trim(Copy(line, n+1, MaxInt));
+
+      if code = 'Type' then
+        FType := value
+      else if code = 'Subtag' then
+        FID := value
+      else if code = 'Description' then
+        FDescription := value;
+    end;
+  end;
+
+  function Escape(s: string): string;
+  begin
+    Result := s.Replace('''', '''''', [rfReplaceAll]);
+  end;
+
+  procedure AddDict(s: TStringList);
+  var
+    i: Integer;
+  begin
+    for i := 0 to s.Count - 1 do
+      FResult.Add('  dict.Add('''+Escape(s.Names[i])+''', '''+Escape(s.ValueFromIndex[i])+''');');
+  end;
+begin
+  FResult := TStringList.Create;
+  FLanguages := TStringList.Create;
+  FScripts := TStringList.Create;
+  FRegions := TStringList.Create;
+  try
+    ParseList;
+    FResult.Add('unit Keyman.System.Standards.BCP47SubtagRegistry;');
```

Comment:
If I understand things correctly, this dynamically builds an updated source code file for consumption/linking to windows/src/global/delphi/general/Keyman.System.LanguageCodeUtils.pas?

Reply:
Correct. Saves distributing the file and parsing each load (and dealing with file paths etc in deployment, which can be a real pain).

Reply:
Is it possible to use this [language data index](https://build.palaso.org/viewType.html?buildTypeId=bt98)? It's what several SIL tools that have language pickers use.

My primary concern is that our set of BCP 47 tags matches everyone else's set of BCP 47 tags.
:/

Reply:
So ... we can, but it's not necessarily as straightforward. Keyman has been using the IANA subtag registry for a while; we can source the libPalaso data if there is a good canonical location for it (teamcity artifacts are really not great for that). It's a separate issue though, not something that has to change with this PR.